nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/DeleteHDFS.java

```diff
@@ -141,20 +141,33 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
             }
 
             Map<String, String> attributes = Maps.newHashMapWithExpectedSize(2);
+            boolean foundMissingFile = false;
             for (Path path : pathList) {
                 attributes.put("filename", path.getName());
                 attributes.put("path", path.getParent().toString());
                 if (fileSystem.exists(path)) {
                     fileSystem.delete(path, context.getProperty(RECURSIVE).asBoolean());
+
                     if (!context.hasIncomingConnection()) {
                         flowFile = session.create();
+                        session.transfer(session.putAllAttributes(flowFile, attributes), REL_SUCCESS);
                     }
-                    session.transfer(session.putAllAttributes(flowFile, attributes), REL_SUCCESS);
+
                 } else {
                     getLogger().warn("File (" + path + ") does not exist");
+
                     if (!context.hasIncomingConnection()) {
                         flowFile = session.create();
+                        session.transfer(session.putAllAttributes(flowFile, attributes), REL_FAILURE);
                     }
+
+                }
+            }
+            if (context.hasIncomingConnection()) {
+                // TODO we only put the last path deleted, change the semantic of the processor?
```

Comment:
We should remove this 'TODO' and instead document the behavior you're describing on the capability decription/and/or the WritesAttribute section.  If it is important for a user to see a listing of deleted items they should not use the wildcard here but rather use ListHDFS to create an explicit set of files and this processor should be honoring precisely the items specified.